### PR TITLE
Config window switcher

### PR DESCRIPTION
**Pull request by johanmalm**
@Flrian - review appreciated
...and @Consolatis of course :smile:

### CODE REVIEW

**Comment by Consolatis** [comment]
Other than ending the first line of the deprecation logs with `[dot][space]` this looks good to me.
I assume that is also what the CodeStyleCheck error is about, that thing is amazingly smart sometimes.

**Comment by johanmalm** [comment]
^ But I want the space there!!

**Comment by Consolatis** [comment]
I think that is what the message is about, the last 2 are missing the dot space. The first message uses it.

**Comment by johanmalm** [comment]
> I think that is what the error is about, the last 2 are missing the dot-space. The first message uses it (and doesn't raise a codecheck error).

Yep. I'll add a space on the next line instead. `checkpatch.pl` is too clever :smile:

**Comment by Consolatis** [comment]
> to be more consistent with the other options already there?

In regard to `rc.xml.all` I agree. I think the docs are fine though.
The parser implementation supports both variants so whatever the user decides to use is fine.

**Comment by johanmalm** [comment]
I've stripped out `window` but left the rest as it is. Ready to merge from my perspective.

**Comment by Consolatis** [comment]
You want to keep the more compact variant in `rc.xml.all` vs the long form with own tags instead of attributes shown above?

File: src/config/rcxml.c [hunk 1]
```diff
@@ -442,12 +442,29 @@ entry(xmlNode *node, char *nodename, char *content)
 		rc.snap_edge_range = atoi(content);
 	} else if (!strcasecmp(nodename, "topMaximize.snapping")) {
 		rc.snap_top_maximize = get_bool(content);
+
+	/* <windowSwitcher show="" windowPreview="" windowOutlines="" /> */
+	} else if (!strcasecmp(nodename, "show.windowSwitcher.core")) {
+		rc.cycle_view_osd = get_bool(content);
+	} else if (!strcasecmp(nodename, "windowPreview.windowSwitcher.core")) {
+		rc.cycle_preview_contents = get_bool(content);
+	} else if (!strcasecmp(nodename, "windowOutlines.windowSwitcher.core")) {
```
Comment (Consolatis) [inline src/config/rcxml.c:451]: These checks include the `window` prefix.

Comment (johanmalm) [reply src/config/rcxml.c]: Whooa. Thanks for catching.

**Comment by johanmalm** [comment]
> You prefer the more compact variant in `rc.xml.all` vs the long form with own tags instead of attributes shown above?

Yes, I think it's neater. Keeps it more compact.

The rest of `rc.xml.all` contains a mix anyway, so I think decision criteria ought to be (in this order):

1. Keep consistent with Openbox 3.6
2. If there is a reasonable chance of wanting to expand with further nested options in future, then use nested `<elements>`
3. Keep succinct - by using `attributes=""`

In this case (1) doesn't apply and I think (2) is pretty unlikely.

**Comment by johanmalm** [comment]
Okay. Ready to go from my angle, but will hang on merging for a day or so in case anyone has any further thoughts as it's been quite a lengthy perspective already :smile:

Will also be a good check of the new website manpage workflow.

**Comment by Consolatis** [comment]
PR looks good. I'll go for the merge.

> Will also be a good check of the new website manpage workflow.

Indeed.

**Pull request merged**